graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +39,340 @@ def __init__(self, query_ast):
+
+    Args:
+        query_node: SubQueryNode, whose list of child query connections may be modified to include
+                    children
+        ast: type Node, the AST that we are trying to split into child components. It is not
+             modified by this function
+        parent_selections: List[Node], containing all property fields (and possibly some vertex
+                           fields) in the level of selections that contains the input ast. It
+                           is not modified by this function
+        type_info: TypeInfo, used to get information about the types of fields while traversing
+                   the query ast
+        edge_to_stitch_fields: Dict[Tuple(str, str), Tuple(str, str)], mapping
+                               (type name, vertex field name) to
+                               (source field name, sink field name) used in the @stitch directive
+                               for each cross schema edge
+        intermediate_out_name_assigner: IntermediateOutNameAssigner, object used to generate
+                                        and keep track of names of newly created @output
+                                        directives
+
+    Returns:
+        Node object to replace the input AST in the selections one level above. If the output
+        is unchanged from the input AST, the exact same object will be returned
+    """
+    # Check if there is a split here. If so, split AST, make child query node, return property
+    # field. If not, recurse on all child selections (if any)
+    if isinstance(ast, Field):
+        parent_type_name = type_info.get_parent_type().name
+        edge_field_name = ast.name.value
+        if (parent_type_name, edge_field_name) in edge_to_stitch_fields:
+            # parent_field_name and child_field_name are names of property fields in the stitch
+            parent_field_name, child_field_name = \
+                edge_to_stitch_fields[(parent_type_name, edge_field_name)]
+            # Get parent field with existing directives
+            parent_property_field = _get_property_field(
+                parent_selections, parent_field_name, ast.directives
+            )
+            # Add @output if needed, record out_name
+            parent_output_name = _get_out_name_optionally_add_output(
+                parent_property_field, intermediate_out_name_assigner
+            )
+            # Create child query node around ast
+            child_query_node, child_output_name = _get_child_query_node_and_out_name(
+                ast, type_info, child_field_name, intermediate_out_name_assigner
+            )
+            # Create and add QueryConnections
+            _add_query_connections(
+                query_node, child_query_node, parent_output_name, child_output_name
+            )
+            # Return parent property field used in the stitch, to be added in selections above
+            return parent_property_field
```

Comment:
Since this ends in a `return`, it might be a good idea to extract everything after the `if` on line 91 into a function, so that we get something nice and simple and fairly obvious like:
```python
if (parent_type_name, edge_field_name) in edge_to_stitch_fields:
    return foo()
```